### DISCUSSION TITLE
Internal Server Error after Login

### DISCUSSION THREAD

**Discussion by Gandalf145** [netbox-community/netbox#8559] [Q&A] [answered]
From the login screen I am receiving Internal Server Error after logging in with the correct super user credentials. 

![Netbox Error](https://user-images.githubusercontent.com/98188026/153283150-a60581a3-6ba0-42ed-882e-993748dc966a.PNG)

- v3.1.6
- Ubuntu 20.04.3 LTS

**Comment by Gandalf145** [comment]
Bump

**Comment by candlerb** [comment, marked as answer]
Sorry, but you have not provided any information which would allow us to help you diagnose the problem.  At the moment, all we can say is "there's something wrong with your installation".

At very least, you need to provide the exact error message or screenshot; the version of Netbox you're running and the OS you're running it under; and what settings you've made in configuration.py (especially anything you've changed relating to authentication).

Also, describe exactly how to reproduce the error.  It sounds to me like Netbox displays its front page correctly, and when you click Login you are prompted for a username and password correctly, but you only get the error *after* you've done that.  Is that correct?

If this error is a Python exception, then it's also very helpful if you can provide the python backtrace.  You can get this by setting EMAIL and ADMINS in configuration.py, and then it will be E-mailed to you; or you can set DEBUG = True and you'll get it in the web interface.

**Reply by Gandalf145** [reply]
Hi candlerb, thank you for the response. I've updated my original post with the info you mentioned and added it below.

- Netbox v3.1.6
- Running on Ubuntu 20.04.03 LTS
- You are correct in that the login page displays correctly and I only receive the error after logging in
- Below are the required settings of the configuration.py file

ALLOWED_HOSTS = ['*']

DATABASE = {
    'NAME': 'name_omitted',         # Database name
    'USER': 'name_omitted',               # PostgreSQL username
    'PASSWORD': 'password_removed',           # PostgreSQL password
    'HOST': 'localhost',      # Database server
    'PORT': '',               # Database port (leave blank for default)
    'CONN_MAX_AGE': 300,      # Max database connection age
}
 
REDIS = {
    'tasks': {
        'HOST': 'localhost',
        'PORT': 6379,
        # Comment out `HOST` and `PORT` lines and uncomment the following if using Redis Sentinel
        # 'SENTINELS': [('mysentinel.redis.example.com', 6379)],
        # 'SENTINEL_SERVICE': 'netbox',
        'PASSWORD': '',
        'DATABASE': 0,
        'SSL': False,
        # Set this to True to skip TLS certificate verification
        # This can expose the connection to attacks, be careful
        # 'INSECURE_SKIP_TLS_VERIFY': False,
    },
    'caching': {
        'HOST': 'localhost',
        'PORT': 6379,
        # Comment out `HOST` and `PORT` lines and uncomment the following if using Redis Sentinel
        # 'SENTINELS': [('mysentinel.redis.example.com', 6379)],
        # 'SENTINEL_SERVICE': 'netbox',
        'PASSWORD': '',
        'DATABASE': 1,
        'SSL': False,

**Reply by candlerb** [reply]
Hmm.  This isn't a problem I've seen before: normally Netbox doesn't start at all if there's a problem.

"Internal Server Error" is probably from the proxy in front (Apache or Nginx) saying it can't get an answer.  But it could be anything, unless your proxy server logs an error (e.g. `/var/log/apache2/error.log` for Apache under Ubuntu).

I'd suggest you get the backtrace as mentioned before.  Also, try running Netbox with "[runserver](https://netbox.readthedocs.io/en/stable/installation/3-netbox/#test-the-application)" so that it bypasses the front-end proxy entirely:

```
cd /opt/netbox
. venv/bin/activate
python3 netbox/manage.py runserver 0.0.0.0:8000 --insecure
```

then point web browser at http://x.x.x.x:8000/.  This may give you a better indication of the error.

**Reply by Gandalf145** [reply]
I checked my nginx logs and there were no errors in the error.log file. 

After following what you advised I found several errors referencing django-auth-ldap. I compared my configuration.py to the configuration.example.py and reverted some changes. I am now able to login. Thank you for your help and direction!
